File: product-service/src/main/java/com/intexsoft/webshop/productservice/producer/ProductEventProducer.java
package com.intexsoft.webshop.productservice.producer;

import com.intexsoft.webshop.messagecommon.event.product.impl.ProductCreatedEvent;
import com.intexsoft.webshop.messagecommon.event.product.impl.ProductDeletedEvent;
import com.intexsoft.webshop.messagecommon.event.product.impl.ProductOrderQuantityIncrementedEvent;
import com.intexsoft.webshop.messagecommon.event.product.impl.ProductUpdatedEvent;

public interface ProductEventProducer {
    void produceProductEventCreated(ProductCreatedEvent productCreatedEvent);

    void produceProductEventDeleted(ProductDeletedEvent productDeletedEvent);

    void produceProductEventUpdated(ProductUpdatedEvent productUpdatedEvent);

    void produceProductOrderQuantityIncremented(ProductOrderQuantityIncrementedEvent productOrderQuantityIncrementedEvent);
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/producer/ProductEventProducer.kt
package com.intexsoft.webshop.productservicekt.producer

import com.intexsoft.webshop.messagecommon.event.product.impl.ProductCreatedEvent
import com.intexsoft.webshop.messagecommon.event.product.impl.ProductDeletedEvent
import com.intexsoft.webshop.messagecommon.event.product.impl.ProductOrderQuantityIncrementedEvent
import com.intexsoft.webshop.messagecommon.event.product.impl.ProductUpdatedEvent

interface ProductEventProducer {
    fun produceProductEventCreated(productCreatedEvent: ProductCreatedEvent)
    fun produceProductEventDeleted(productDeletedEvent: ProductDeletedEvent)
    fun produceProductEventUpdated(productUpdatedEvent: ProductUpdatedEvent)
    fun produceProductOrderQuantityIncremented(productOrderQuantityIncrementedEvent: ProductOrderQuantityIncrementedEvent)
}